08_flask_html_basics.md

```diff
@@ -0,0 +1,592 @@
+
+```html
+<ul>
+    {% for animal in animals %}
+        <li>{{ animal }}</li>
+    {% endfor %}
+</ul>
+```
+
+</details>
+
+## CRUD, HTTP Methods, inputs and forms
+
+_CRUD_ stands for Create, Read, Update, Delete, which are the four basic operations that a web application would perform.
+
+Thinking of your application so far, you have only implemented the ability to Read information upon visiting the `/` path. Sometimes this is enough for simple websites that deliver static content. For more interactive user experiences, however, you will likely need to implement other operations.
+
+_HTTP_ (HyperText Transfer Protocol) stands at the foundation of data communication over the internet, as it standardises the way hypertext documents are served to clients. It is a  stateless request-response protocol based on client-server communication. A typical client could be a browser, which then communicates with servers that deliver HTML files or other media resources.
+
+HTTP defines several operations through which the client communicates with the server. Four of the most common ones are:
+- `GET`: Is used to request information from the client. It is similar to a Read operation of CRUD, and is intended to have no side effects on the server side.
+- `POST`: Is used to request the server to perform a certain operation that will possibly modify its resources, such as uploading a message, creating an object in a database, asking to send an email, etc.
+- `PUT`/`PATCH`: Used for updating resources.
+- `DELETE`: Used for deleting resources.
+
+> While the standard dictates that HTTP methods behave in a certain expected way, the server's code ultimately decides what its methods do. Even so, adhering to these conventions is most of the times desiarable.
+
+This is just a very brief introduction to HTTP and CRUD. These topics will be discused in more detail in further modules.
+
+### Exercise 8.12
+
+Although you haven't explicitly mentioned it in the code, while communicating with the `/` route of your application through the browser, one of the HTTP methods has to be used. Which of the above is most likely to be used?
+
+<details markdown="1"><summary>Hint</summary>
+While the Flask application is running, you can refresh the page and check the terminal in VS Code. The method used, along with the path requested, should appear in the log.
+</details>
+
+After you've figured out the HTTP method, try setting it explicitly in the code. You can do that by adding a `methods` parameter to the app routing, like this:
+
+```python
+@app.route("/", methods=['HTTP_METHOD'])
+```
+
+Refresh the page to check that everything is still working.
+
+### HTTP Response Codes
+
+Since HTTP is a stateless, request-response protocol, the server needs to communicate to the client the status of the request (e.g. whether it was successful or not), in order for the client to interpret the server's response correctly.
```

Comment:
```suggestion
Since HTTP is a request-response protocol, the server needs to communicate to the client the status of the request (e.g. whether it was successful or not), in order for the client to interpret the server's response correctly.
```